### PR TITLE
feat: 🎸 [JIRA:HCPSDKFIORIUIKIT-2700] Object Cell Enhancement

### PR DESCRIPTION
**Pull request by hengyi-zhang**

### CODE REVIEW

File: Apps/Examples/Examples/FioriSwiftUICore/ObjectItem/ObjectCellEnhancementExample.swift [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                            self.isShowAccessoryMenu = true
+                        }
+                    } image: { _ in
+                        FioriIcon.actions.overflow
+                    }
+                    .fioriButtonStyle(FioriTertiaryButtonStyle(colorStyle: .normal).eraseToAnyFioriButtonStyle())
+                }
+            })
+        }
+        .navigationTitle("Object Cell Enhancement")
```
Comment (dyongxu) [inline Apps/Examples/Examples/FioriSwiftUICore/ObjectItem/ObjectCellEnhancementExample.swift:158]:
```suggestion
        .navigationTitle("Object Cell Enhancement for Button")
```

File: Apps/Examples/Examples/FioriSwiftUICore/ObjectItem/ObjectItemExample.swift [hunk 2]
```diff
@@ -31,6 +31,13 @@ struct ObjectItemExample: View {
             {
                 Text("Avatars and Icons Tests")
             }
+            
+            NavigationLink {
+                ObjectCellEnhancementExample()
+            } label: {
+                Text("Object Cell Enhancement")
```
Comment (dyongxu) [inline Apps/Examples/Examples/FioriSwiftUICore/ObjectItem/ObjectItemExample.swift:38]:
```suggestion
                Text("Object Cell Enhancement for Button")
```

File: Apps/Examples/Examples/FioriSwiftUICore/ObjectItem/ObjectCellEnhancementExample.swift [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            }, subtitle: {
+                Text("Three Phase Pad Mounted Transformer (533423)")
+            }, footnote: {
+                Text("1000 - Hamburg, MECHANIK")
+            }, description: {
+                Text("Customer noticed that the transformer started to over heat within 45 minutes each time he turned it on at 7:30am.  The first technician who looked at this did not have the correct additional tools to complete the job.")
+            }, detailImage: {
+                Image("attachment009").resizable().frame(width: 45, height: 45)
+            }, objectItemButton: {
+                FioriButton(title: "Label", action: { _ in
```
Comment (dyongxu) [inline Apps/Examples/Examples/FioriSwiftUICore/ObjectItem/ObjectCellEnhancementExample.swift:22]:
```suggestion
                FioriButton(title: "Buton", action: { _ in
```

File: Apps/Examples/Examples/FioriSwiftUICore/ObjectItem/ObjectCellEnhancementExample.swift [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            }, footnote: {
+                Text("1000 - Hamburg, MECHANIK")
+            }, description: {
+                Text("Customer noticed that the transformer started to over heat within 45 minutes each time he turned it on at 7:30am.  The first technician who looked at this did not have the correct additional tools to complete the job.")
+            }, status: {
+                Text("High")
+            }, detailImage: {
+                Image("attachment009").resizable().frame(width: 45, height: 45)
+            }, objectItemButton: {
+                FioriButton(title: "Label", action: { _ in
```
Comment (dyongxu) [inline Apps/Examples/Examples/FioriSwiftUICore/ObjectItem/ObjectCellEnhancementExample.swift:41]:
```suggestion
                FioriButton(title: "Button", action: { _ in
```

File: Apps/Examples/Examples/FioriSwiftUICore/ObjectItem/ObjectCellEnhancementExample.swift [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            }, description: {
+                Text("Customer noticed that the transformer started to over heat within 45 minutes each time he turned it on at 7:30am.  The first technician who looked at this did not have the correct additional tools to complete the job.")
+            }, status: {
+                Text("High")
+            }, substatus: {
+                Text("Scheduled")
+            }, detailImage: {
+                Image("attachment009").resizable().frame(width: 45, height: 45)
+            }, objectItemButton: {
+                FioriButton(title: "Label", action: { _ in
```
Comment (dyongxu) [inline Apps/Examples/Examples/FioriSwiftUICore/ObjectItem/ObjectCellEnhancementExample.swift:62]:
```suggestion
                FioriButton(title: "Button", action: { _ in
```

**Review by dyongxu — CHANGES_REQUESTED** [review]
1. I suggested some test code change. Pls don't commit that directly from github(It will fail the build since the direct commit does not include a keyword in the commit message)
2. Can you review the screen with Xiaoyu to see if the "overflow" button alignment is correct? (It does not look correct to me)

**Comment by dyongxu** [comment]
Screenshot 2024-11-12 at 15 33 13
@hengyi-zhang @xiaoyu0722 Can you review the overflow button alignment is correct ?

File: Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift [hunk 6]
```diff
@@ -3,7 +3,10 @@ import SwiftUI
 
 /// A view that displays information of an object.
 // sourcery: CompositeComponent
-protocol _ObjectItemComponent: _TitleComponent, _SubtitleComponent, _FootnoteComponent, _DescriptionComponent, _StatusComponent, _SubstatusComponent, _DetailImageComponent, _IconsComponent, _AvatarsComponent, _FootnoteIconsComponent, _FootnoteIconsTextComponent, _TagsComponent, _ActionComponent {}
+protocol _ObjectItemComponent: _TitleComponent, _SubtitleComponent, _FootnoteComponent, _DescriptionComponent, _StatusComponent, _SubstatusComponent, _DetailImageComponent, _IconsComponent, _AvatarsComponent, _FootnoteIconsComponent, _FootnoteIconsTextComponent, _TagsComponent, _ActionComponent {
+    // sourcery: @ViewBuilder
+    var objectItemButton: FioriButton? { get }
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift:8]: What's this button used for?

An action usually should have its own component protocol like `ActionComponent`.

**Review by billzhou0223 — CHANGES_REQUESTED** [review]

File: Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift (hunk 6 above)
Comment (hengyi-zhang) [reply Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift:8]: The objectItemButton for accessory enhancement

File: Apps/Examples/Examples/FioriSwiftUICore/ObjectItem/ObjectCellEnhancementExample.swift (hunk 1 above)
Comment (hengyi-zhang) [reply Apps/Examples/Examples/FioriSwiftUICore/ObjectItem/ObjectCellEnhancementExample.swift]: fixed!

**Comment by xiaoyu0722** [comment]
> @xiaoyu0722 Can you review the overflow button alignment is correct ?

No problem for the alignment.
Spec ask for the trailing alignment, also contain the button margin.
Screenshot 2024-11-13 at 14 01 53

**Review by dyongxu — APPROVED** [review]

**Comment by dyongxu** [comment]
@billzhou0223 We need to sync up with SAPFiori 24.12 for main codeline. Is there some requested change here? Or it is ok to merge ?

**Pull request merged**